supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalBackend.scala

```diff
@@ -1,29 +1,227 @@
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
 
-class LocalBackend extends BackendJobExecutionActor {
-  /**
-    * Restart or resume a previously-started job.
-    */
-  override def recover: Future[BackendJobExecutionResponse] = ???
+object LocalBackend {
+  val logger = LoggerFactory.getLogger("LocalBackend")
+  // TODO Support GCS ?
+  val fileSystems = List(FileSystems.getDefault)
+
+  private def splitFqn(fullyQualifiedName: FullyQualifiedName): (String, String) = {
+    val lastIndex = fullyQualifiedName.lastIndexOf(".")
+    (fullyQualifiedName.substring(0, lastIndex), fullyQualifiedName.substring(lastIndex + 1))
+  }
+
+  case class Command(argv: Seq[String]) {
+    override def toString = argv.map(s => "\"" + s + "\"").mkString(" ")
+  }
+}
+
+class LocalBackend(override val jobDescriptor: BackendJobDescriptor,
+                   override val configurationDescriptor: BackendConfigurationDescriptor) extends BackendJobExecutionActor with SharedFileSystem {
+
+  import LocalBackend._
+  import better.files._
+  import cromwell.core.PathFactory._
+
+  import scala.sys.process._
+
+  // Mutable process variable assigned when the execute method is called.
+  // Needs to be accessible to be killed when the job is aborted.
+  private var process: Option[Process] = None
+
+  val workflowDescriptor = jobDescriptor.descriptor
+  val jobPaths = new JobPaths(workflowDescriptor, backendConfiguration, jobDescriptor.key)
+  val fileSystemsConfig = backendConfiguration.getConfig("filesystems")
+  override val sharedFsConfig = fileSystemsConfig.getConfig("local")
+
+  val call = jobDescriptor.key.call
+  val callEngineFunction = {
+    val callContext = new CallContext(
+      jobPaths.callRoot.toString,
+      jobPaths.stdout.toAbsolutePath.toString,
+      jobPaths.stderr.toAbsolutePath.toString
+    )
+
+    new LocalCallEngineFunctions(fileSystems, callContext)
+  }
+
+  val evaluator = {
+    val declarations = workflowDescriptor.workflowNamespace.workflow.declarations ++ call.task.declarations
+    val unqualifiedWorkflowInputs = workflowDescriptor.inputs map {
+      case (fqn, v) => splitFqn(fqn)._2 -> v
+    }
+    val knownInputs = unqualifiedWorkflowInputs ++ jobDescriptor.symbolMap
+    val lookup = WdlExpression.standardLookupFunction(knownInputs, declarations, callEngineFunction)
+    new ValueEvaluator(lookup, callEngineFunction)
+  }
+
+  private def evaluate(wdlExpression: WdlExpression) = evaluator.evaluate(wdlExpression.ast)
+
+  // TODO determine how this plays with the init/validation work
+  val runtimeAttributes = {
+    val evaluateAttrs = call.task.runtimeAttributes.attrs mapValues evaluate
+    // Fail the call if runtime attributes can't be evaluated
+    val evaluatedAttributes = TryUtils.sequenceMap(evaluateAttrs, "Runtime attributes evaluation").get
+    LocalRuntimeAttributes(evaluatedAttributes)
+  }
+
+  lazy val runsOnDocker = runtimeAttributes.dockerImage.isDefined
```

Comment:
TOL: It would be super-neat if we could have `LocalDocker` as a separate backend:
- To avoid all of this messy decision making in a single BE file. 
- To make extending to RemoteDocker backends much easier
- It isn't a big deal now that we can mix and match Docker and Non-docker backends in the same workflow now
- It'd be an awesome early test case for PBE (since the FS is genuinely shared, we won't have to worry about localisation/delocalisation)
- It would make test cases more focused
- This code file is smaller and simpler

(Genuine TOL - if you disagree or want to merge this PR first, no problem)